department-of-veterans-affairs/abd-vro
# Claim Tracker: Create application

## Summary

Creates the new 

Associated tickets or Slack threads:
- #3362 

## How does this fix it?[^1]

* creates the base skeleton of the claim tracker application
* creates the rest endpoint to add claims to track
* creates the database table for `claims.tracked_claims`
* sets up code quality checks using ruff, isort, mypy, and runs black formatting
* creates unit tests for health check endpoint and track claim endpoint

## How to test this PR
- download branch and cd to root dir
- run `./gradlew :domain-bie-events:svc-claim-tracker:check`
- run `./gradlew clean docker`
- run `COMPOSE_PROFILES="platform" ./gradlew :dockerComposeUp`
- run `COMPOSE_PROFILES="all" ./gradlew :domain-bie-events:dockerComposeUp`
- send curl: 
```
curl -L 'http://localhost:8150/track/v1/claim' -H 'Content-Type: application/json' -d '{
    "claim_id": 123,
    "established_at": "2024-01-01T00:00:00Z",
    "feature_name":"feature",
    "feature_enabled":true
}'
```
[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
